pkl/evaluator_options.go

```diff
@@ -453,6 +453,7 @@ var WithProjectEvaluatorSettings = func(project *Project) func(opts *EvaluatorOp
 			for scheme, reader := range evaluatorSettings.ExternalModuleReaders {
 				opts.ExternalModuleReaders[scheme] = ExternalReader(reader)
 				if evaluatorSettings.AllowedModules == nil { // if no explicit allowed modules are set in the project, allow declared external module readers
+					WithDefaultAllowedModules(opts)
 					opts.AllowedModules = append(opts.AllowedModules, scheme+":")
 				}
 			}
```

Comment:
This mimics Pkl CLI behavior. If the project specifies external module/resource readers but not explicit allowed modules/resources then the schemes for the external readers are allowed.

This produces a problem because `NewProjectEvaluator*` loads project evaluator setting before other evaluator options, which means that when `MaybePreconfiguredOptions` is used its `WithDefaultAllowed*` options are never invoked.

To account for this, this change explicitly pulls in the defaults when the project defines no allowed modules/resources and does define external readers.